Update flasher-related parameters (flasher, laser, led, illuminators)

GernotMaier opened the pull request:
**Very early PR to allow experts to keep an eye on the changes**

Review of all parameters and merging of duplication:

- `flasher_photons` replaces `laser_photons`, `led_photons`, `photons_per_run`


Closes #1664

GernotMaier commented on src/simtools/model/array_model.py:
```diff
@@ -214,40 +227,78 @@ def _build_telescope_models(self, site_model: SiteModel, array_elements: dict) -
             if name not in exported_models:
                 self._logger.debug(f"Exporting configuration file for telescope {name}")
-                tel_model.write_sim_telarray_config_file()
+                tel_model.write_sim_telarray_config_file(
+                    additional_models=self.calibration_models.get(tel_model.name)
+                )
```
This is safe - as both dicts are generated in the same loop.

GernotMaier commented on .github/workflows/CI-unittests.yml:
```diff
@@ -114,7 +114,8 @@ jobs:
           cd database_scripts/
-          ./upload_from_model_repository_to_db.sh ${{ env.SIMTOOLS_DB_SIMULATION_MODEL }} ${{ env.SIMTOOLS_DB_SIMULATION_MODEL_VERSION }}
+          # TODO temporary from branch for testing of flasher parameters
+          ./upload_from_model_repository_to_db.sh ${{ env.SIMTOOLS_DB_SIMULATION_MODEL }} ${{ env.SIMTOOLS_DB_SIMULATION_MODEL_VERSION }} flaser_models
```
I know - just read the PR text.

GernotMaier commented on .github/workflows/CI-integrationtests.yml:
```diff
@@ -112,7 +112,8 @@ jobs:
           cd database_scripts/
-          ./upload_from_model_repository_to_db.sh ${{ env.SIMTOOLS_DB_SIMULATION_MODEL }} ${{ env.SIMTOOLS_DB_SIMULATION_MODEL_VERSION }}
+          # TODO temporary from branch for testing of flasher parameters
+          ./upload_from_model_repository_to_db.sh ${{ env.SIMTOOLS_DB_SIMULATION_MODEL }} ${{ env.SIMTOOLS_DB_SIMULATION_MODEL_VERSION }} flaser_models
```
I know - just read the PR text.

GernotMaier commented on src/simtools/simtel/simtel_config_writer.py:
```diff
@@ -104,22 +106,81 @@ def write_telescope_config_file(self, config_file_path, parameters, telescope_na
+
+        simtel_par.update(dict.fromkeys(mapping.values(), 0.0))
+        if shape in mapping:
+            simtel_par[mapping[shape]] = width
```
Good pint. Fixed.

tobiaskleiner commented on src/simtools/model/array_model.py:
```diff
@@ -214,40 +227,78 @@ def _build_telescope_models(self, site_model: SiteModel, array_elements: dict) -
+
+        return telescope_models, calibration_models
+
+    def _build_calibration_models(self, telescope_model, site_model, calibration_device_types):
```
Ok this is good then

tobiaskleiner commented on src/simtools/utils/geometry.py:
```diff
@@ -236,3 +237,36 @@ def transform_ground_to_shower_coordinates(x_ground, y_ground, z_ground, azimuth
+
+    Assumes definition of shapes as in 'camera_body_shape' model parameter:
+
+    - circle: shape = 0, width is diameter
```
Ok all good, just wanted to be sure I understand here the use with hexagon and square.

GernotMaier commented:
@tobiaskleiner - I think I've addressed all your points, thanks again! Let me know if there is anything else to do or if I can merge this into the flasher-refactor branch.

GernotMaier commented:
There is a complete example, what would you like me to add? 

```

2. Simulate at a configurable position (1km above array center) and pointing downwards:

    .. code-block:: console

        simtools-simulate-illuminator --light_source ILLN-01 \
        --light_source_position 0. 0. 1000. \
        --light_source_pointing 0. 0. -1. \
        --telescope MSTN-15 --site North \
        --model_version 6.0.0
```

tobiaskleiner commented:
> There is a complete example, what would you like me to add?
> 
> ```
> 
> 2. Simulate at a configurable position (1km above array center) and pointing downwards:
> 
>     .. code-block:: console
> 
>         simtools-simulate-illuminator --light_source ILLN-01 \
>         --light_source_position 0. 0. 1000. \
>         --light_source_pointing 0. 0. -1. \
>         --telescope MSTN-15 --site North \
>         --model_version 6.0.0
> ```

But this one here is not how we want it and it wouldn't work, as it doesnt point towards the telescope: Here the light source is not above the telescope, but at the array center. 
Would be nice if you could specify that here and use the telescope position of MSTN-15 (xy) and 1000 as height for the light source.

tobiaskleiner reviewed (approved):
Thanks @GernotMaier, can be merged.
